graphql/schema.graphql

```diff
@@ -90,6 +90,80 @@ type Mutation {
   Edit schedule and Plug and Play Compnay 3's and 4's Algorithms
   """
   editCourse(input: EditCourseInput): Response!
+    @deprecated(reason: "Use updateSchedule instead")
+  """
+  Update an entire schdule.
+  """
+  updateSchedule(input: UpdateScheduleInput!): UpdateScheduleResponse!
+}
+
+input UpdateScheduleInput {
+  """
+  ID of the schedule to update. If not given, the current schedule will be updated.
+  """
+  id: ID
+  """
+  The updated courses
+  """
+  courses: [CourseSectionInput!]!
+  """
+  Whether to perform validation on the backend through algorithm 1.
+  """
+  skipValidation: Boolean!
+  """
+  Which algorithm to use. If COMPANY4 is selected then validation will not be performed regardless of skipValidation.
+  """
+  validation: Company!
+}
+
+input CourseSectionInput {
+  """
+  The course identifier
+  """
+  id: CourseUpdateInput!
```

Comment:
Can this be CourseID so that it matches the CourseSection type that's currently in the schema?

Reply:
GraphQL doesn't let us use a `type` as an `input` which is why there's a new definition with the exact same fields.